week11_homework/app.js

```diff
@@ -0,0 +1,147 @@
+        player.moveRight();
+    }
+    if (keydownEvent.code === "ArrowLeft") {
+        player.moveLeft();
+    }
+    if (keydownEvent.code === "ArrowUp") {
+        player.moveUp();
+    }
+    if (keydownEvent.code === "ArrowDown") {
+        player.moveDown();
+    }
+
+})
+
+//cream o lista unde adaugam monstrii, din lista asta vom luam pe rand monstrii pentru a le creea un setInterval
+//prin care sa se miste
+const listMonster = []
+
+//-cream monstrii + adaugam monstrii pe pozitii random in container + adaugam in listMonster[]
+for (let i = 0; i < 30; i++) {
+    const monster = new Monster(gameContainer);
+    listMonster.push(monster);
+}
+
+//TODO
+//Optional Homework:   
+//Adaugam tabelul de vieti HTML si numarul de vieti
+const tabelVietiHtml = document.getElementById("tabel-vieti")
+const arrayVieti = []
+for(let i = 1; i < 4; i++){
+    const life = `life${i}`;
+    arrayVieti.push(life)
+
+    const newLife = document.createElement("span")
+    newLife.innerHTML = life;
+    tabelVietiHtml.appendChild(newLife)
+}
+
+//6 + Optional homework - Daca functia monsterTouchPlayer() este adevarata 
+//sa ii scada o viata, daca nu are ce scadea, sa fie game over. La fiecare atingere
+//player-ul are imunitate 3 secunde, timp in care nu poate fi atins de monstrii.
+var gameOverValue = false;
+var loseLife = setInterval(function gameOver(){
+    if(monsterTouchPlayer() === true){
+        if(gameOverValue === true){
+            alert("GAME OVER");
+            clearInterval(loseLife);
+            clearInterval(randomMonsterMoves);
+        } 
+        else {
+            arrayVieti.pop()
+                if(arrayVieti.length <= 1){
+                    gameOverValue = true;
+                }
+                console.log("player lose 1 life")
+
+            clearInterval(loseLife)
+            console.log("player-ul este imun 3 secunde")
+            setTimeout(() => {
+                loseLife = setInterval(gameOver(), 100);
+            }, 3000);
+        }
+    }
+},100)
+
+//6.Functie sa verific daca player-ul a atins vreun monstru
+
+function monsterTouchPlayer (){
+    let sumTouches = false;
+    for (let i = 0; i < listMonster.length; i++) {
+        
+        const monsterElement = listMonster[i].element;
+    
+        const positionPlayerTop = parseInt(player.element.style.top);
+        const positionPlayerLeft = parseInt(player.element.style.left);
+        const positionMonsterTop = parseInt(monsterElement.style.top);
+        const positionMonsterLeft = parseInt(monsterElement.style.left);
+
+    
+        if (positionMonsterTop >= positionPlayerTop && positionMonsterTop <= (positionPlayerTop + 30) &&
+            positionMonsterLeft >= positionPlayerLeft && positionMonsterLeft <= (positionPlayerLeft + 30)) {
+                console.log("monster touch player");
+                sumTouches = sumTouches || true;
+        } else{
+            sumTouches = sumTouches || false;
+        }
+        return sumTouches
```

Comment:
trebui inafara for-ului

Reply:
cred ca de asta nu merge bine. As recomanda sa faci fara setInterval verificare si sa o pui dupa ce sa schimba pozitia la fiecare monstru